spec/lib/alaveteli_localization_spec.rb

```diff
@@ -0,0 +1,78 @@
+# -*- encoding : utf-8 -*-
+require 'spec_helper'
+
+describe AlaveteliLocalization do
+
+  context '.set_locales' do
+
+    before do
+      AlaveteliLocalization.set_locales('en_GB es', 'en_GB')
+    end
+
+    it 'sets FastGettext.locale' do
+      expect(FastGettext.locale).to eq("en_GB")
+    end
+
+    it 'sets FastGettext.default_locale' do
+      expect(FastGettext.default_locale).to eq("en_GB")
+    end
+
+    it 'sets FastGettext.default_available_locales' do
+      expect(FastGettext.default_available_locales).to eq([:en_GB, :es])
```

Comment:
The mistake there may be checking `available_default_locales` instead of `default_locales` - one returns an array of strings, the other an array of symbols.

(*wait*  as these are already handily formatted as symbols they're **returned as-is** as `AlaveteliLocalization.available_locales` so maybe this test is fine but would benefit from a comment. And I would benefit from some more coffee. So not a stupid question, no.)

To answer a broader question, it seems to be that `FastGettext` returns strings whereas `I18n` returns symbols, which is why historically we were casting everything to string (or to symbol) before taking any action to avoid type errors as it was hard to know where `locale` had come from at any given point.

```
2.3.0 :001 > FastGettext.default_available_locales
 => [:en, :es]
2.3.0 :002 > FastGettext.available_locales
 => ["en", "es"]
2.3.0 :003 > FastGettext.locale
 => "en"
2.3.0 :004 > I18n.locale
 => :en
```

From https://github.com/grosser/gettext_i18n_rails/blob/v1.1.0/spec/gettext_i18n_rails_spec.rb#L61-L64

```
it "links FastGettext with I18n locale" do
  FastGettext.locale = 'xx'
  I18n.locale.should == :xx
end
```